tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -1498,8 +1571,10 @@ private[emitter] class JSDesugaring(semantics: Semantics,
           } else {
             val names = fields map (_._1)
             val items = fields map (_._2)
-            unnest(items) { (newItems, env) =>
-              redo(JSObjectConstr(names.zip(newItems)))(env)
+            unnestPropertyNames(names) { (newNames, env) =>
+              unnest(items) { (newItems, env) =>
```

Comment:
Hum ... we need to check what is the proper evaluation order of keys and values in ES6, in the presence of computed names. Hopefully it interleaves: key0, name0, key1, name1, etc. If that's the case the above way of unnesting is not correct, as it might evaluate some keys earlier than required.